variant-remapping-automation/genome_target_tracker.py

```diff
@@ -0,0 +1,96 @@
+from argparse import ArgumentParser
+
+import requests
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query
+from ebi_eva_common_pyutils.taxonomy.taxonomy import get_scientific_name_from_ensembl
+
+logger = logging_config.get_logger(__name__)
+logging_config.add_stdout_handler()
+
+remapping_genome_target_table = 'evapro.remapping_genome_target_tracker'
+ensembl_url = 'http://rest.ensembl.org/info/assembly'
+
+
+def get_all_taxonomies_from_eva(private_config_xml_file):
+    taxonomy_list = []
+    with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+        query = 'SELECT DISTINCT taxonomy_id FROM evapro.taxonomy'
+        for taxonomy in get_all_results_for_query(pg_conn, query):
+            taxonomy_list.append(taxonomy[0])
+
+    return taxonomy_list
+
+
+def get_tax_asm_from_eva(private_config_xml_file):
+    eva_tax_asm = {}
+    with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+        query = f"""SELECT DISTINCT taxonomy_id, assembly_id FROM {remapping_genome_target_table} 
+        WHERE current_previous='current'"""
+        for tax_id, asm in get_all_results_for_query(pg_conn, query):
+            eva_tax_asm[tax_id] = asm
+
+    return eva_tax_asm
+
+
+def get_tax_asm_from_ensembl(taxonomy_list):
+    ensembl_tax_asm = {}
+    for tax_id in taxonomy_list:
+        try:
+            logger.info(f'Query Ensembl for species name using taxonomy {tax_id}')
+            sp_name = get_scientific_name_from_ensembl(tax_id)
```

Comment:
The [previous script](https://github.com/EBIvariation/eva-tools/blob/master/variant-remapping-automation/gather_release_species.py#L83) also queried NCBI for scientific name if Ensembl didn't return anything, should we have that here as well?

Reply:
If Ensembl is not able to return species name for a taxonomy, I would assume the assembly information also will not be present for that taxonomy in Ensembl, since we query that using species name itself.
Please correct me, if you feel that's not the case and the 2 are mutually exclusive in Ensembl.

Reply:
Based on what Tim said this morning, this should be fine (though we might later add new sources for species names as well as supported assemblies) :+1: